apps/src/javalab/JavalabEditor.jsx

```diff
@@ -774,6 +778,14 @@ class JavalabEditor extends React.Component {
               />
             </div>
             <Tab.Content id="tab-content" animation={false}>
+              {isReadOnlyWorkspace && hasOpenCodeReview && isViewingOwnProject && (
```

Comment:
is it ever possible for `isReadOnlyWorkspace` and `isViewingOwnProject` to be true and `hasOpenCodeReview` to be false? I think this logic makes sense here, but am wondering if there's more banners we should be adding for another case, or if we should just put a comment that the only time you have a read only project that's your own you are in code review.

Reply:
Yeah I think you're right (an open code review is the only way you get read only if you're looking at your own project). I didn't even have the `isReadOnlyWorkspace` flag here originally, but thought I should add it since the message explicitly talks about read only. I'll add a comment.

Reply:
LMK if this sounds right -- only show this banner if you're looking at the most recent version of your project? Is there an easy way to know if you're looking at most recent version?

Reply:
The version url param will be present if you're looking at an older version